WordPress/gutenberg
# [RNMobile] Performance improvements in File related controls

**Merge after `Video` PR: https://github.com/WordPress/gutenberg/pull/28220**

## Description

Part of: https://github.com/WordPress/gutenberg/pull/28196

Contains changes only related to the `File` block and controls available within block settings and inherit changes from `Video` PR.

| BLOCK			| KIND OF SETTING		| CONTROL COMPONENT		| NESTED CONTROL COMPONENT	|
| :---			| :---: 				|  :---: 				| :---:  					|
| File			| File block Settings	| `SelectControl`		| 							|
|				|						| `ToggleControl`		| 							|	
|				|						| `ToggleControl`		| 							|	
|				|						| `TextControl`			| 							|	

## How has this been tested?

Test manually `File` block on web and mobile.

You can paste the snippet to have `File` block filled:

```

<div class="wp-block-file"><a href="https://callstackhappiryuhome.files.wordpress.com/2020/12/pexels-photo-3732527.jpg">pexels-photo-3732527</a><a href="https://callstackhappiryuhome.files.wordpress.com/2020/12/pexels-photo-3732527.jpg" class="wp-block-file__button" download>Download</a></div>

```


_or_

1. Open post on web (run gutenberg locally)
2. Add `File` block and attach file
3. Play with settings and observe if work properly
4. Enable all of them and save post
5. Open post on mobile
6. Play with settings

**Expect everything works correctly**

## Screenshots 

### RESULTS

**1. pressing copy url button**

- updating time reduced from **7.06ms** to **3.49ms**
- avoid redundant updates in other controls

before | after
--- | ---
<img width="1123" alt="Screenshot 2021-01-15 at 11 20 33" src="https://user-images.githubusercontent.com/22746080/104713821-aeeae180-5724-11eb-878a-7bf1058357d6.png">| <img width="1123" alt="Screenshot 2021-01-15 at 11 19 33" src="https://user-images.githubusercontent.com/22746080/104713808-abeff100-5724-11eb-8933-499a4a3ab4b9.png">

## Types of changes





Performance improvements

## Checklist:
